data_safe_haven/exceptions/__init__.py

```diff
@@ -1,50 +1,50 @@
-class DataSafeHavenException(Exception):
+class DataSafeHavenError(Exception):
     pass
 
 
-class DataSafeHavenCloudException(DataSafeHavenException):
+class DataSafeHavenCloudError(DataSafeHavenError):
     pass
 
 
-class DataSafeHavenConfigException(DataSafeHavenException):
+class DataSafeHavenConfigError(DataSafeHavenError):
     pass
 
 
-class DataSafeHavenInputException(DataSafeHavenException):
+class DataSafeHavenInputError(DataSafeHavenError):
     pass
 
 
-class DataSafeHavenInternalException(DataSafeHavenException):
+class DataSafeHavenInternalError(DataSafeHavenError):
     pass
 
 
-class DataSafeHavenIPRangeException(DataSafeHavenException):
+class DataSafeHavenIPRangeError(DataSafeHavenError):
     pass
 
 
-class DataSafeHavenNotImplementedException(DataSafeHavenInternalException):
+class DataSafeHavenNotImplementedError(DataSafeHavenInternalError):
     pass
 
 
-class DataSafeHavenParameterException(DataSafeHavenException):
+class DataSafeHavenParameterError(DataSafeHavenError):
     pass
 
 
-class DataSafeHavenSSLException(DataSafeHavenException):
+class DataSafeHavenSSLError(DataSafeHavenError):
     pass
 
 
-class DataSafeHavenAzureException(DataSafeHavenCloudException):
+class DataSafeHavenAzureError(DataSafeHavenCloudError):
     pass
 
 
-class DataSafeHavenUserHandlingException(DataSafeHavenInternalException):
+class DataSafeHavenUserHandlingError(DataSafeHavenInternalError):
     pass
 
 
-class DataSafeHavenMicrosoftGraphException(DataSafeHavenAzureException):
+class DataSafeHavenMicrosoftGraphError(DataSafeHavenAzureError):
     pass
 
 
-class DataSafeHavenPulumiException(DataSafeHavenCloudException):
+class DataSafeHavenPulumiError(DataSafeHavenCloudError):
     pass
```

Comment:
Worth checking whether all these errors are currently used in the code. Some may be unnecessary at the moment.

Reply:
Definitely worth doing. Do you know if there is a tool to check for unused exceptions?

Again, feels out of scope for the PR?

Reply:
I think a quick "grep" will do the job here.

Reply:
These are all used :+1:.

I have opened an issue about clarifying their use https://github.com/alan-turing-institute/data-safe-haven/issues/1518